oracle/service.go

```diff
@@ -208,38 +230,76 @@ func (s *oracleSvc) processSetPriceFeed(ticker, providerName string, pricePuller
 		"provider": pricePuller.ProviderName(),
 	})
 
-	ctx := context.Background()
 	symbol := pricePuller.Symbol()
 
 	t := time.NewTimer(5 * time.Second)
 	for {
 		select {
 		case <-t.C:
-			ctx, cancelFn := context.WithTimeout(ctx, maxRespTime)
+			ctx, cancelFn := context.WithTimeout(context.Background(), maxRespTime)
 			defer cancelFn()
+			// define price and asset pair to tracking late
+			var err error
+			price := zeroPrice
+			var assetPairs []*oracletypes.AssetPair
+
+			if pricePuller.OracleType() == oracletypes.OracleType_Stork {
+				storkPricePuller, ok := pricePuller.(*storkPriceFeed)
+				if !ok {
+					metrics.ReportFuncError(s.svcTags)
+					feedLogger.WithFields(log.Fields{
+						"oracle":  "Stork Oracle",
+						"retries": maxRetriesPerInterval,
+					}).WithError(err).Errorln("can not convert to stork price feed")
+					continue
+				}
+				assetPairs, err = storkPricePuller.PullAssetPairs(s.storkWebsocket)
+				if err != nil {
+					metrics.ReportFuncError(s.svcTags)
+					feedLogger.WithError(err).Warningln("retrying PullAssetPairs after error")
+
+					for i := 0; i < maxRetriesPerInterval; i++ {
+						if assetPairs, err = storkPricePuller.PullAssetPairs(s.storkWebsocket); err != nil {
+							time.Sleep(time.Second)
+							continue
+						}
+						break
+					}
 
-			result, err := pricePuller.PullPrice(ctx)
-			if err != nil {
-				metrics.ReportFuncError(s.svcTags)
-				feedLogger.WithError(err).Warningln("retrying PullPrice after error")
+					if err != nil {
+						metrics.ReportFuncError(s.svcTags)
+						feedLogger.WithFields(log.Fields{
+							"retries": maxRetriesPerInterval,
+						}).WithError(err).Errorln("failed to fetch asset pairs")
 
-				for i := 0; i < maxRetriesPerInterval; i++ {
-					if result, err = pricePuller.PullPrice(ctx); err != nil {
-						time.Sleep(time.Second)
+						t.Reset(pricePuller.Interval())
 						continue
 					}
-					break
 				}
-
+			} else {
```

Comment:
We can skip pushing an empty update to the channel


```suggestion
				if len(assetPairs) == 0 {
					t.Reset(pricePuller.Interval())
					continue
				}
			} else {
```